### DISCUSSION TITLE
No JRE 20 found when trying to build a spring boot image with maven plugin

### DISCUSSION THREAD

**Discussion by qassiel** [paketo-buildpacks/feedback#217] [Java Team]
Hello. I'm building a Java 20 Spring Boot 3.1.4 image with spring-boot-maven-plugin using paketo-buildpacks/java. It worked in the past, but today I got an error saying that no valid JRE was available.
![build error](https://github.com/paketo-buildpacks/feedback/assets/7278271/b07622f2-636c-426a-be72-eb7ae2c300aa)

I found out that the latest version of  paketobuildpacks/bellsoft-liberica (10.3.0) drops support for JRE/JDK 20. Any reason you didn't include Java 20 support in the buildpack?

I managed to make it work downgrading the buildpack to version 10.2.9 in the pom.xml. 
```
...
<buildpacks>
    <buildpack>paketobuildpacks/bellsoft-liberica:10.2.9</buildpack>
    <buildpack>urn:cnb:builder:paketo-buildpacks/java</buildpack>
</buildpacks>
...
```
Is this the right approach to deal with this issue? 

Unfortunately we won't be able to upgrade to Java 21 until Spring Boot supports it (maybe at the end of november).

Best regards,
Martin Medina

**Comment by dmikusa** [comment]
> I found out that the latest version of paketobuildpacks/bellsoft-liberica (10.3.0) drops support for JRE/JDK 20. Any reason you didn't include Java 20 support in the buildpack?

Correct. Java 20 is basically EOL with the release of Java 21. We only ship what is supported upstream.

Your solution is to use an LTS release which would be Java 17 or Java 21. If you can't do either of those, then what you're doing is the correct workaround. You can downgrade to the last buildpack which includes Java 20 support. You won't be missing any dependency updates because there will be no more upstream Java 20 updates.

**Reply by qassiel** [reply]
Thanks for your answer, Daniel. We'll try to downgrade to Java 17 until Spring Boot is ready for Java 21.
